src/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java

```diff
@@ -105,21 +115,39 @@ public void importMetadataDatasource( String domainId, InputStream metadataFile,
     publish( pentahoSession );
   }
 
-  public RepositoryFileAclDto getMetadataAcl( String domainId ) throws PentahoAccessControlException {
-    if ( !canAdministerCheck() ) {
-      throw new PentahoAccessControlException();
+  public RepositoryFileAclDto getMetadataAcl( String domainId )
+      throws PentahoAccessControlException, FileNotFoundException {
+    checkMetadataExists( domainId );
+    if ( aclMetadataHelper != null ) {
+      final RepositoryFile domainRepositoryFile = aclMetadataHelper.getDomainRepositoryFile( domainId );
+      final RepositoryFileAcl acl = aclMetadataHelper.getAclHelper().getAclFor( domainRepositoryFile );
```

Comment:
A couple things. 

I don't like getAclHelper(). These methods should be on the ACL Aware interfaces directly.

getAclFor( RepositoryFile):
Move this to the ACLAware interface and change it to take the domainId String if possible. I don't like fetching the RepositoryFile and passing it back in, the interface should use the logical DomainID wherever possible